binders/kafka-binder/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/BatchWithDlqCustomizerDisablesBinderDlqTests.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.kafka;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.binder.kafka.properties.KafkaConsumerProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Soby Chacko
+ */
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE, properties = {
+	"spring.cloud.function.definition=consumer1",
+	"spring.cloud.stream.kafka.bindings.consumer1-in-0.consumer.enable-dlq=true"})
+@EmbeddedKafka
+public class BatchWithDlqCustomizerDisablesBinderDlqTests {
+
+	@Autowired
+	private KafkaMessageChannelBinder kafkaMessageChannelBinder;
+
+	@Test
+	void batchWithDlqCustomizerDisablesBinderDlq() {
+		KafkaConsumerProperties kafkaConsumerProperties =
+			kafkaMessageChannelBinder.getExtendedConsumerProperties("consumer1-in-0");
+		ExtendedConsumerProperties<KafkaConsumerProperties> extendedConsumerProperties =
+			new ExtendedConsumerProperties<>(kafkaConsumerProperties);
+		extendedConsumerProperties.setBatchMode(true);
+
+		MessageHandler errorMessageHandler =
+			kafkaMessageChannelBinder.getErrorMessageHandler(null, null, extendedConsumerProperties);
+		// verifies that binder does not create a message handler for errors, which otherwise creates a handler for DLQ.
+		assertThat(errorMessageHandler).isNull();
+	}
+
+	@SpringBootApplication
```

Comment:
It is dangerous to use this annotation in tests. It comes with `@ComponentScan`, so unexpected beans might appear in the context from the same package as this test class. Consider to use `@EnableAutoConfiguration` and `@Configuration` pair instead . You can see them from the mentioned `@SpringBootApplication`

Reply:
Thanks, Artem, for spotting that. I was copying from an existing test.